### PR TITLE
Implemented procedures list

### PR DESCRIPTION
**Pull request by Trinovantes**
Can't wait till we hit those JavaScript race conditions

I would suggest reading the commits separately. Tests will come after the port is finished (or someone else can write them).

### CODE REVIEW

**Comment by Trinovantes** [comment]
screenshot from 2015-11-04 17 31 48
screenshot from 2015-11-04 17 31 28

File: src-backbone/app/js/views/procedures/account_navbar_view.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        Marionette.ItemView.prototype.constructor.call(this, options);
+    },
+
+    templateHelpers: function() {
+        var self = this;
+
+        return {
+            username: function() {
+                // TODO fetch from server
+                return 'Trinovantes';
```
Comment (Ommy) [inline src-backbone/app/js/views/procedures/account_navbar_view.js]: So I guess this will be a separate ticket?

**Comment by Ommy** [comment]
Overall looks fine. I just don't like how the colour for the background saying you don't have any procedures is the same as the list of procedures. I think if a user has no procedures, something else should be there, not a list element. Maybe a static image saying they don't have procedures, or simply a http://getbootstrap.com/components/#alerts.

**Comment by Trinovantes** [comment]
Maybe yellow? Then again, it's just a note and not a warning or error. I could try making the procedures lighter colour

File: src-backbone/app/js/views/procedures/account_navbar_view.js (hunk 1 above)
Comment (Trinovantes) [inline src-backbone/app/js/views/procedures/account_navbar_view.js]: Yea we need an `/api/account` endpoint for getting info and changing email/passwords

**Comment by Ommy** [comment]
I guess maybe that bootstrap link is not the best suggestion, but there should be some difference between an empty procedure list and a populated one.

**Comment by Trinovantes** [comment]
make procedures blue?

**Comment by Ommy** [comment]
Yeah maybe that'll work. Try it out. Just some difference between the two will be good.

**Comment by Trinovantes** [comment]
The blues look too harsh under the grey text. I decided to just keep it as a white bordered card with rounded corners.

screenshot from 2015-11-06 17 51 46

**Comment by Ommy** [comment]
Is an empty list of procedures still the gray box?

**Comment by Trinovantes** [comment]
Yea

**Comment by Trinovantes** [comment]
screenshot from 2015-11-09 17 19 27
screenshot from 2015-11-09 17 19 12

**Comment by Ommy** [comment]
New design looks slick, +1

File: src-backbone/app/js/models/procedures_collection.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        return (this._sort_asc ? 1 : -1) * this._asc_comparator(a.get(this._comparator_key), b.get(this._comparator_key));
+    },
+
+    set_comparator_key: function(key) {
+        log.info('Procedure Comparator Key: ' + key);
+        this._comparator_key = key;
+    },
+
+    set_order_asc: function(sort_asc) {
+        log.info('Procedure Sort Asc: ' + sort_asc);
```
Comment (ConnorCimowsky) [inline src-backbone/app/js/models/procedures_collection.js]: Do we want to keep these log statements?

File: src-backbone/app/js/models/procedures_model.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module.exports = Backbone.Model.extend({
+
+    defaults: {
+        'title': 'Untitled', // TODO allow nullable titles
+        'author': 'Stephen', // TODO either fetch user's username or let API allow nullable authors
+        'pages': [],
+    },
+
+    parse: function(response, options) {
+        var parse_date_in_response = function (key) {
```
Comment (ConnorCimowsky) [inline src-backbone/app/js/models/procedures_model.js]: epic nit but I don't think the space is intentional in `function (key)`

**Comment by ConnorCimowsky** [comment]
LGTM :1:

File: src-backbone/app/js/models/procedures_collection.js (hunk 2 above)
Comment (Trinovantes) [inline src-backbone/app/js/models/procedures_collection.js]: These output to the browser's console to help us debug. I don't think we actually need to keep these in production (maybe we can replace it with analytics events?)

**Pull request merged**